packages/tools/src/utilities/cine/playClip.ts

```diff
@@ -262,4 +296,143 @@ function _stopClipWithData(playClipData) {
   }
 }
 
+function _getVolumeFromViewport(viewport): Types.IImageVolume {
+  const actorEntry = viewport.getDefaultActor();
+
+  if (!actorEntry) {
+    console.warn('No actor found');
+  }
+
+  const volumeId = actorEntry.uid;
+  return cache.getVolume(volumeId);
+}
+
+function _createStackViewportCinePlayContext(
+  viewport: StackViewport
+): CINETypes.CinePlayContext {
+  const imageIds = viewport.getImageIds();
+
+  return {
+    get numScrollSteps(): number {
+      return imageIds.length;
+    },
+    get currentStepIndex(): number {
+      return viewport.getTargetImageIdIndex();
+    },
+    get frameTimeVectorEnabled(): boolean {
+      // It is always in acquired orientation
+      return true;
+    },
+    scroll(delta: number): void {
+      scroll(viewport, { delta, debounceLoading: debounced });
+    },
+  };
+}
+
+function _createVolumeViewportCinePlayContext(
+  viewport: VolumeViewport,
+  volume: Types.IImageVolume
+): CINETypes.CinePlayContext {
+  const actorEntry = viewport.getDefaultActor();
+
+  if (!actorEntry) {
+    console.warn('No actor found');
```

Comment:
shouldn't it return or something?

Reply:
Changed it to return `undefined`. Is there a way to have an `enabledElement` with no volume/stack?